userspace/libsinsp/CMakeLists.txt

```diff
@@ -306,41 +313,25 @@ if(NOT DEFINED SINSP_AGENT_CGROUP_MEM_PATH_ENV_VAR)
 endif()
 add_definitions(-DSINSP_AGENT_CGROUP_MEM_PATH_ENV_VAR="${SINSP_AGENT_CGROUP_MEM_PATH_ENV_VAR}")
 
-# Build our pkg-config "Libs:" flags. For now, loop over SINSP_PKGCONFIG_LIBRARIES. If
-# we ever start using pkg_search_module or pkg_check_modules in cmake/modules
-# we could add each module to our "Requires:" line instead. We might need to
-# expand this to use some of the techniques in
-# https://github.com/curl/curl/blob/curl-7_84_0/CMakeLists.txt#L1539
 set(SINSP_PKG_CONFIG_LIBS)
 set(SINSP_PKG_CONFIG_LIBDIRS "")
-foreach(sinsp_lib ${SINSP_PKGCONFIG_LIBRARIES})
-	if(${sinsp_lib} MATCHES "^-")
-		# We have a flag. Pass it through unchanged.
-		list(APPEND SINSP_PKG_CONFIG_LIBS ${sinsp_lib})
-	elseif(${sinsp_lib} MATCHES "/")
-		# We have a path. Convert it to -L<dir> + -l<lib>.
-		get_filename_component(sinsp_lib_dir ${sinsp_lib} DIRECTORY)
-		list(APPEND SINSP_PKG_CONFIG_LIBDIRS -L${sinsp_lib_dir})
-		get_filename_component(sinsp_lib_base ${sinsp_lib} NAME_WE)
-		string(REGEX REPLACE "^lib" "" sinsp_lib_base ${sinsp_lib_base})
-		list(APPEND SINSP_PKG_CONFIG_LIBS -l${sinsp_lib_base})
-	elseif(${sinsp_lib} STREQUAL "scap")
-		# We require libscap.pc, so skip it.
-	else()
-		# Assume we have a plain library name. Prefix it with "-l".
-		list(APPEND SINSP_PKG_CONFIG_LIBS -l${sinsp_lib})
-	endif()
-endforeach()
-
-# Build our pkg-config "Cflags:" flags.
-set(SINSP_PKG_CONFIG_INCLUDES "")
-foreach(sinsp_include_directory ${LIBSINSP_INCLUDE_DIRS})
-	list(APPEND SINSP_PKG_CONFIG_INCLUDES -I${sinsp_include_directory})
-endforeach()
-
-string(REPLACE ";" " " SINSP_PKG_CONFIG_LIBS "${SINSP_PKG_CONFIG_LIBS}")
-list(REMOVE_DUPLICATES SINSP_PKG_CONFIG_LIBDIRS)
-string(REPLACE ";" " " SINSP_PKG_CONFIG_LIBDIRS "${SINSP_PKG_CONFIG_LIBDIRS}")
-list(REMOVE_DUPLICATES SINSP_PKG_CONFIG_INCLUDES)
-string(REPLACE ";" " " SINSP_PKG_CONFIG_INCLUDES "${SINSP_PKG_CONFIG_INCLUDES}")
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/libsinsp.pc.in ${CMAKE_CURRENT_BINARY_DIR}/libsinsp.pc @ONLY)
+
+list(REMOVE_DUPLICATES SINSP_PKGCONFIG_LIBRARIES)
+list(REMOVE_DUPLICATES SINSP_PKGCONFIG_REQUIRES)
+list(REMOVE_DUPLICATES SINSP_PKGCONFIG_REQUIRES_PRIVATE)
+
+file(GENERATE
+	OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/libsinsp.pc
+	CONTENT "prefix=${CMAKE_INSTALL_PREFIX}
+libdir=\${prefix}/${CMAKE_INSTALL_LIBDIR}
+includedir=\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}
+
+Name: libsinsp
+Description: lib for System INSPection
+Version: ${FALCOSECURITY_LIBS_VERSION}
+
+Requires: libscap $<JOIN:${SINSP_PKGCONFIG_REQUIRES}, >
+Requires.private: $<JOIN:${SINSP_PKGCONFIG_REQUIRES_PRIVATE}, >
+Libs: -L\${libdir} -lsinsp $<JOIN:${SINSP_PKG_CONFIG_LIBS}, -l>
```

Comment:
Did you mean SINSP_PKGCONFIG_LIBRARIES here?

Reply:
Good catch, though we would also need to guard against SINSP_PKGCONFIG_LIBRARIES being empty, perhaps?

Reply:
The reason I missed that is that I was using the wrong variable to start with (it should have been `SINSP_PKGCONFIG_LIBRARIES`).

Reply:
> Good catch, though we would also need to guard against SINSP_PKGCONFIG_LIBRARIES being empty, perhaps?

Done, with: `Libs: -L\${libdir} -lsinsp $<BOOL:${SINSP_PKGCONFIG_LIBRARIES}:-l$<JOIN:${SINSP_PKGCONFIG_LIBRARIES}, -l>>`